core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -78,9 +78,11 @@ public String partition() {
   private boolean failMissingDeletePaths = false;
   private int duplicateDeleteCount = 0;
   private boolean caseSensitive = true;
+  private boolean allDeletesReferenceManifests = true;
 
   // cache filtered manifests to avoid extra work when commits fail.
   private final Map<ManifestFile, ManifestFile> filteredManifests = Maps.newConcurrentMap();
+  private final Set<String> manifestsWithDeletedFiles = Sets.newConcurrentHashSet();
```

Comment:
Does this have to be `ConcurrentHashSet`? Why not `HashSet`?

Reply:
Good point, no it doesn't need to be a ConcurrentHashSet. We are not modifying this set of manifests concurrently, we are only reading from it concurrently during the filtering which will only happen after it's modified.